src/wallet/wallet2.cpp

```diff
@@ -1384,10 +1387,14 @@ void wallet2::expand_subaddresses(const cryptonote::subaddress_index& index)
     for (; index2.minor < end; ++index2.minor)
     {
        const crypto::public_key &D = pkeys[index2.minor - begin];
-       m_subaddresses[D] = index2;
+       new_elements.push_back(std::make_pair(D, index2));
     }
     m_subaddress_labels[index.major].resize(index.minor + 1);
   }
+
+  std::sort(new_elements.begin(), new_elements.end(), [](const spair_t &e0, const spair_t &e1) { return e0.first < e1.first; });
+  boost::container::ordered_unique_range_t tag;
+  m_subaddresses.insert(tag, new_elements.begin(), new_elements.end());
```

Comment:
Doing this speeds it up a LOT compared to adding one by one. I can't recall whether I tried that insert version, I'll try it.

Reply:
Lots slower. So much I ^Ced. This is my test code:

<pre>
#if 0
  std::sort(v.begin(), v.end());
  boost::container::ordered_unique_range_t tag;
  flat_map.insert(tag, v.begin(), v.end());
#else
  flat_map.insert(v.begin(), v.end());
#endif
</pre>

Reply:
Yeah, I thought about this while away from my computer. The problem is that the map cannot "touch" the iterator range, and therefore has to search over the entire internal vector every insert instead of a decreasing range.

The `adopt_sequence` method should be faster in all cases, and doing a pre-sort shouldn't be helpful.

Reply:
Looks like the adopt_sequence was added in 1.65, so I'd rather not use it given the inserts are typically one off and faster than the current anyway.

Reply:
Too bad this was a recent addition to Boost. I don't know what you meant by "faster than the current anyway" - handing the memory buffer to `flat_map` is the fastest possible way to initialize the data structure.

Reply:
I meant insertion in the flat_map is faster than the current std::unordered_map insertion is.